### PR TITLE
Add teamId to enable org-wide tokens usage

### PR DESCRIPTION
**Pull request by lucaghersi**
To support org-wide apps, you need to be able to pass "team_id" when a org token is used.
Not all APIs support this parameter (https://api.slack.com/enterprise/apps/changes-apis)

This PR add support for teamId where possible, while making some rational changes to some APIs, for example using Request Objects where parameters list growth too much.

This PR contains breaking contract changes.

That said, I will be more than happy to make changes / adaptation / whatever you see fit.
Open for feedback :)

### CODE REVIEW

**Comment by lucaghersi** [comment]
BTW, I tried to contain changes to the minimum amount possible, but there are potential improvements I didn't apply to maintain consistency. For example primary constructors, adding guards for required values, etc.
It's even possible to drop this changes I made alltogether, and make some kind of "global teamId parameter" in the construction of the web client, and support teamId directly.
As I said, happy to receive feedbacks.

**Comment by stoiveyp** [comment]
Hi @lucaghersi - haven't forgotten about this, just had a busy couple of days and as it's a large number of changes I want time to look through it properly before I merge it.

Really appreciate you've made changes that keep in the style of the API, I haven't spotted anything I have a problem with - hopefully it'll be approved and published in the next 24hrs 🤞

**Comment by lucaghersi** [comment]
No rush and thanks for looking at it. What I thought (next PR maybe?) is that it would be possible to remove the "accessToken" and "teamId" from all methods and put them in the Api client constructor and make them global, massively simplifying all methods.

**Comment by stoiveyp** [comment]
Not sure what you mean about accessToken? We set the token at the client level - you don't pass it into every method?

As for teamId. I want the current API Client to be a clear representation of the API as much as possible - so no global state outside of the token. But - there's definitely the possibility of something like `client.ForTeam(teamId)` that could then return something like an `ITeamSlackClient` with methods that have the parameter removed 💭

File: Slack.NetStandard/WebApi/IReactionsApi.cs [hunk 1]
```diff
@@ -7,11 +7,8 @@ public interface IReactionsApi
     {
         Task<WebApiResponse> Add(string channel, Timestamp timestamp, string name);
         Task<ReactionGetResponse> Get(string channel, Timestamp timestamp, bool? full = null);
-
-        Task<ReactionListResponse> List(string user);
```
Comment (stoiveyp) [inline Slack.NetStandard/WebApi/IReactionsApi.cs:11]: Would you be able to move this to a ReactionListRequest object in the same way you did with the IUsersApi - moving to a request object model for those calls with Cursor + Limit options makes more sense moving forward, to more easily accommodate future changes.

File: Slack.NetStandard/WebApi/TeamApi.cs [hunk 2]
```diff
@@ -14,45 +14,14 @@ public TeamApi(IWebApiClient client)
...
-       }
-
-        public Task<TeamAccessLogResponse> AccessLogs(int count, int page)
+        public Task<TeamAccessLogResponse> AccessLogs(TeamAccessLogRequest request)
         {
-            return AccessLogs(null, count, page);
+            return _client.MakeUrlEncodedCall<TeamAccessLogResponse>("team.accessLogs", request);
         }
 
-        public Task<TeamAccessLogResponse> AccessLogs(long? before, int? count, int? page)
```
Comment (stoiveyp) [inline Slack.NetStandard/WebApi/TeamApi.cs:27]: The `list` methods have always been annoying as there are standard patterns we use for those - so don't mind having a request objects as replacements.

In this instance there's no advantage to removing the simple methods, so let's keep those working please (don't mind if they use your request object method to be implemented, but let's keep them in the interface)

**Review by stoiveyp — CHANGES_REQUESTED** [review]
Couple of minor changes to the implementation - but once those are done it'll get merged in as the next major release. This is really good stuff, thank you!

File: Slack.NetStandard/WebApi/TeamApi.cs (hunk 2 above)
Comment (lucaghersi) [reply Slack.NetStandard/WebApi/TeamApi.cs:27]: Done. I restored the previous contract with the optional team_id parameter for all of them.

File: Slack.NetStandard/WebApi/IReactionsApi.cs (hunk 1 above)
Comment (lucaghersi) [reply Slack.NetStandard/WebApi/IReactionsApi.cs:11]: Done!

**Review by stoiveyp — APPROVED** [review]

**Pull request merged**